test/distributed/pipelining/test_schedule_multiproc.py

```diff
@@ -682,16 +683,67 @@ def test_pipeline_schedule_runtime_custom_sched(self, ScheduleClass):
     @requires_nccl()
     @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
     @parametrize(
-        "schedule_class", [ScheduleVShaped, ScheduleUnbalanced, ScheduleZBVZeroBubble]
+        "schedule_class",
+        [ScheduleZBVZeroBubble, ScheduleDualPipeV],
+    )
+    @parametrize("use_new_runtime", [False, True])
+    def test_v_shape_schedules(self, schedule_class, use_new_runtime):
+        n_stages = 4
+        rank_stages = {0: [0, 3], 1: [1, 2]}
```

Comment:
might be a good idea to test n_stages = 6 or 8, so there is at least one rank that is both sender/recvr.  (there are behavior differences between beginning, middle, and end stages right?

Reply:
Makes sense, all the tests currently only use world_size=2, but will update them.